src/plugins/share/public/types.ts

```diff
@@ -112,32 +118,39 @@ export interface ShareMenuItemV2 extends ShareMenuItemBase {
   helpText?: ReactElement;
   copyURLButton?: { id: string; dataTestSubj: string; label: string };
   generateExportButton?: ReactElement;
-  generateExport: (args: {
-    intl: InjectedIntl;
-    optimizedForPrinting?: boolean;
-  }) => Promise<unknown>;
+  /**
+   * Function to trigger an export
+   */
+  generateExport: (args: ScreenshotExportOpts) => Promise<unknown>;
+  /**
+   * Function to generate a URL to be used for automating export
+   * Not applicable for exports that do not call a remote API (i.e Lens CSV export)
+   */
+  generateExportUrl?: (args: ScreenshotExportOpts) => string | undefined;
   theme?: ThemeServiceSetup;
   renderLayoutOptionSwitch?: boolean;
   layoutOption?: 'print';
-  absoluteUrl?: string;
   generateCopyUrl?: URL;
   renderCopyURLButton?: boolean;
 }
 
-export type ShareMenuItem = ShareMenuItemLegacy | ShareMenuItemV2;
+export interface ShareMenuProviderV2 {
+  readonly id: string;
+  getShareMenuItems: (context: ShareContext) => Array<Omit<ShareMenuItemV2, 'intl'>>;
+}
+interface ShareMenuProviderLegacy {
+  readonly id: string;
+  getShareMenuItemsLegacy: (context: ShareContext) => ShareMenuItemLegacy[];
+}
```

Comment:
After untangling `ShareMenuItemLegacy` from this area of code, it appears that nothing is actually using the legacy interface except for example code and test code. This should probably be cleaned up in a later PR.